frame/assets/src/functions.rs

```diff
@@ -450,30 +450,35 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 			return Ok(())
 		}
 
-		Self::can_increase(id, beneficiary, amount, true).into_result()?;
-		Asset::<T, I>::try_mutate(id, |maybe_details| -> DispatchResult {
+		Self::can_increase(id.clone(), beneficiary, amount, true).into_result()?;
+		Asset::<T, I>::try_mutate(&id, |maybe_details| -> DispatchResult {
 			let details = maybe_details.as_mut().ok_or(Error::<T, I>::Unknown)?;
 			ensure!(details.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
 			check(details)?;
 
-			Account::<T, I>::try_mutate(id, beneficiary, |maybe_account| -> DispatchResult {
-				match maybe_account {
-					Some(ref mut account) => {
-						account.balance.saturating_accrue(amount);
-					},
-					maybe_account @ None => {
-						// Note this should never fail as it's already checked by `can_increase`.
-						ensure!(amount >= details.min_balance, TokenError::BelowMinimum);
-						*maybe_account = Some(AssetAccountOf::<T, I> {
-							balance: amount,
-							reason: Self::new_account(beneficiary, details, None)?,
-							status: AccountStatus::Liquid,
-							extra: T::Extra::default(),
-						});
-					},
-				}
-				Ok(())
-			})?;
+			Account::<T, I>::try_mutate(
+				id.clone(),
```

Comment:
```suggestion
				&id,
```
I hope clippy finds those.

Reply:
It doesn't, that's the problem. :P